### ISSUE TITLE
Adv Loss is not supported by the paper???

### ISSUE THREAD

**Issue by xljhtq** [FrankWork/fudan_mtl_reviews#3]
Hi,  I want to know where the adv loss is different from the domain loss?? 
In another word,   the adv loss in  the paper "Adversarial Multi-task Learning for Text Classification"   has not described clearly. So i want to know what the equation is??

**Comment by FrankWork** [comment]
total_loss = task_loss + adv_loss + diff_loss + l2_loss

**Comment by xljhtq** [comment]
@FrankWork  In your code, "total_loss = task_loss + adv_loss + diff_loss + l2_loss" ,   to minimize the total_loss, then the adv_loss will be decreasing. But in reality, we should let adv_loss increase in order to get the shared feature. 
So what I should do to maximize the adv_loss or minimize the adv_loss ???

**Comment by FrankWork** [comment]
there is a function `flip_gradient` to maximize the adv_loss

**Comment by ammaarahmad1999** [comment]
Hi, do you know equivalent function of flip_gradient in pytorch
